pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingData.java

```diff
@@ -130,8 +132,97 @@ public long skip(final long n) {
             long numSkipped = in.skip(clamped);
             position += numSkipped;
             return numSkipped;
-        } catch (IOException e) {
+        } catch (final IOException e) {
+            throw new DataAccessException(e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public long readBytes(@NonNull final byte[] dst, final int offset, final int maxLength) {
+        if (maxLength < 0) {
+            throw new IllegalArgumentException("Negative maxLength not allowed");
+        }
+        if (!hasRemaining()) {
+            return 0;
+        }
+        try {
+            int bytesRead = in.read(dst, offset, maxLength);
+            position += bytesRead;
+            if (bytesRead < maxLength) {
+                final int nextByte = tryReadNextByte();
+                if (nextByte != -1) {
+                    dst[offset + bytesRead] = (byte) nextByte;
+                    bytesRead++;
+                }
+            }
+            return bytesRead;
+        } catch (final IOException e) {
             throw new DataAccessException(e);
         }
     }
+
+    @Override
+    public long readBytes(@NonNull final ByteBuffer dst) {
+        if (!dst.hasArray()) {
+            return ReadableSequentialData.super.readBytes(dst);
+        }
+        if (!hasRemaining()) {
+            return 0;
+        }
+        final byte[] dstArr = dst.array();
+        final int dstArrOffset = dst.arrayOffset();
+        final int dstPos = dst.position();
+        final long len = Math.min(remaining(), dst.remaining());
+        try {
+            int bytesRead = in.read(dstArr, dstPos + dstArrOffset, Math.toIntExact(len));
```

Comment:
If the intent is to read all the requested data if possible, and block when needed, we should use `readNBytes` instead of read.  The "read a single byte" workaround isn't needed if we do that.

Reply:
Changed to use `readNBytes()`